packages/di/src/lib/injector.ts

```diff
@@ -44,15 +44,25 @@ export class Injector {
 
     // check for a provider definition
     if (provider) {
-      return this.#createAndCache<T>(provider.use);
+      if (provider.use) {
+        const use = provider.use;
+
+        return this.#createAndCache<T>(token, () => new use());
+      } else if (provider.factory) {
+        const facotory = provider.factory;
+
+        return this.#createAndCache<T>(token, facotory);
+      } else {
+        throw new Error(`Provider for ${token.name} found but is missing either 'use' or 'factory'`)
+      }
```

Comment:
I technically think it will work by just falling back to the token, but I am thinking I prefer it to be more explicit.

Reply:
aye, when configuring a provider, one properly needs to do so!

minor: might want to consider prettier to add semicolons automatically